lib/kube/proxy/forwarder.go

```diff
@@ -1921,6 +1921,7 @@ func (f *Forwarder) serializedRequestClientCreds(authContext authContext) (*tls.
 	// cancel == nil means that another request is in progress, so simply wait until
 	// it finishes or fails
 	f.log.Debugf("Another request is in progress for %v, waiting until it gets completed.", authContext)
+	// REVIEW(gavin) potential race here because ctx may be derived from f.ctx in `f.getOrCreateRequestContext`
```

Comment:
Maybe?

```suggestion
	// TODO(gavin): potential race here because ctx may be derived from f.ctx in `f.getOrCreateRequestContext`
```

Reply:
these comments were just for my own bookkeeping to include in the PR

Reply:
I'm not sure I follow. Do we intend to commit the comments or are we going to address them here?

If they are planned follow ups I'd rather have TODOs, I don't know what REVIEW is supposed to mean.

Reply:
I just wanted to discuss them here, and I left the notes inline to track other places I saw this. Did not intend to commit them.